Gh mirror

armintoepfer opened the pull request.

evolvedmicrobe commented:
Is this needed for swig binding?

nlhepler commented:
No, it's because const members cause all sorts of issues with copy assignment and so on. I should've fixed this when moving things around.

dalexander committed:
Stub out a section for MB to write in the ZScore math details.

nlhepler committed:
more stringent with alpha/beta mismatch at first

so we can kick out reads with borderline behavior
before they cause issues during polishing

dalexander committed:
Add detail to the history and model sections; add some images

armintoepfer committed:
Change submodule urls to bitbucket

dalexander committed:
Flesh out some details on numerics, and add detail on Integrator classes

armintoepfer committed:
Add pbcopper submodule

Switch CLI to pbcopper

- Extract ConsensusSettings
- Unify CLI option setup in one function
- Remove cpp-optparse
- Update tests, a shortened form of the actual option has been used that is not valid with pbcopper.
- Don’t build pbcopper docs and examples
- Outsource PlainOption.h
- Implement ConsensusSettings in its own cpp file
- Minimal documentation
- Add changes to CHANGELOG, inc version

armintoepfer committed:
Merge pull request #2 in SAT/unanimity from bugfix/SAT-131-change-submodules to master

* commit '3a7095d8b53f4eb33ca09beec7241ae54cf3d135':
  Change submodule urls to bitbucket

Commit:
Merge pull request #3 in SAT/unanimity from strictEarlyTol to master

* commit '2b93f32ed44662634572644f07d91d04b21e24fa':
  more stringent with alpha/beta mismatch at first

nlhepler committed:
ccs does not do short-form options

Commit:
Merge pull request #4 in SAT/unanimity from noShortForm to master

* commit 'c1b4e8cd80532bdff6984556d2cfdf0ddf526711':
  ccs does not do short-form options

mpsbpbi committed:
added zscore text and image to this branch.

mpsbpbi committed:
added zscore and image to this branch

armintoepfer committed:
Code documentation

Commit:
Merge pull request #5 in SAT/unanimity from feature/SAT-90-add-section-on-zscore-math to master

* commit '8f0ca56a375d4c7fffd1c49a0b14c14b498e4632':
  added zscore and image to this branch
  added zscore text and image to this branch.

bnbowman committed:
Merge branch 'design-document' of ssh://bitbucket.nanofluidics.com:7999/sat/unanimity

Conflicts:
	doc/ConsensusCore2-DesignAndImplementation/ZScoreMath.rst

Commit:
Merge pull request #7 in SAT/unanimity from SAT-168 to master

* commit 'd5f57713421c1e82139d4d6ab579546258b059fe':
  Add pbcopper submodule

nlhepler committed:
do not crash when a dirent is without '.'

armintoepfer committed:
Merge pull request #6 in SAT/unanimity from SAT-13 to master

* commit '3cced321ce1297f833451c60eb9c15afb91689fe':
  Code documentation

armintoepfer committed:
Change PlainOption namespace to Data

armintoepfer committed:
Merge pull request #11 in SAT/unanimity from namespace-fix to master

* commit '08e16c012257899caeec08f20c66c77d4d6b20af':
  Change PlainOption namespace to Data

armintoepfer committed:
Restore cpp-optparse

Needed for current pblaa support

armintoepfer committed:
Merge pull request #12 in SAT/unanimity from restore-cppoptparse to master

* commit '560098f23dc8022c96d1c589b0cc443da788071b':
  Restore cpp-optparse

Commit:
Merge pull request #10 in SAT/unanimity from fixModelLoading to master

* commit '3e948458916675d2628ae75d6fda5f3ee372d796':
  do not crash when a dirent is without '.'

dalexander committed:
Merge pull request #9 in SAT/unanimity from bugfix/SAT-91-add-section-showing-garden-path-use-of-main-classes to master

* commit '27f94e4123affd49b5c38b0f873a06345cc2151f':
  Flesh out some details on numerics, and add detail on Integrator classes
  Add detail to the history and model sections; add some images
  Stub out a section for MB to write in the ZScore math details.

mpsbpbi committed:
updated ZScoreMath.rst to be correctly Sphinx renderable

dalexander committed:
Provide access to the matrices, in different "conventions", via Evaluator::{AlphaView, BetaView}

dalexander committed:
Merge pull request #14 in SAT/unanimity from bugfix/SAT-90-zscore-formatting to master

* commit '5c4275731583e5e0bb2bb7bb50a9ac9427698538':
  updated ZScoreMath.rst to be correctly Sphinx renderable

dalexander committed:
Require concordant feature lengths in Read constructor

dalexander committed:
Address Lance's concerns---eliminate lingering cruft

Commit:
Merge pull request #15 in SAT/unanimity from diagnostics-merge to master

* commit '0dd7e14c187aa5ab3679a8d35c5d9e43419f2f0b':
  Address Lance's concerns---eliminate lingering cruft
  Provide access to the matrices, in different "conventions", via Evaluator::{AlphaView, BetaView}

dalexander committed:
Fix tests that were failing to create Read objects correctly

Incorrect uses of std::vector constructor, typos, etc

dalexander committed:
Read constructor: Make error message more informative

Commit:
Merge pull request #16 in SAT/unanimity from read-check-feature-lengths to master

* commit '068a7cd3fa38a77b7e5ee1125e6d5a4d8ff792e3':
  Read constructor: Make error message more informative
  Fix tests that were failing to create Read objects correctly
  Require concordant feature lengths in Read constructor

bnbowman committed:
Adding PoaGraph CSV output for debug

armintoepfer committed:
Fix boost version

armintoepfer committed:
Merge pull request #19 in SAT/unanimity from boost-docs to master

* commit '88078b8978569a1acfdb769bbbc0cf35f5c81ae0':
  Fix boost version

armintoepfer committed:
Explain ccs make target explicitly

armintoepfer committed:
Clarify help in README

armintoepfer committed:
Fix namespace comments

dalexander committed:
Merge pull request #20 in SAT/unanimity from namespace-comment-fixes to master

* commit '9bff44016e78dc1188638caeee5a40bc20ca5d6f':
  Fix namespace comments

dalexander committed:
Merge pull request #21 in SAT/unanimity from install-docs to master

* commit '7fced311747dcf21e0df8bca3baeb0c53c84a231':
  Clarify help in README
  Explain ccs make target explicitly

armintoepfer committed:
Switch to relative submodules

bnbowman committed:
Fix POA CSV code Formatting

dalexander committed:
Merge pull request #17 in SAT/unanimity from poa_graph_csv to master

* commit 'bd73437195ddfdb1d0d616e28c12e85d1c13bf71':
  Fix POA CSV code Formatting
  Adding PoaGraph CSV output for debug

armintoepfer committed:
Add bamboo build and test script

This is a 1:1 copy of the bamboo build

armintoepfer committed:
Merge pull request #23 in SAT/unanimity from bamboo-script to master

* commit 'd71890c40aae0d7bc62a71dcba712e0058528028':
  Add bamboo build and test script

armintoepfer committed:
Simplify retrieval of external git repos

armintoepfer committed:
Merge pull request #24 in SAT/unanimity from relative-submodules to master

* commit '6ab4cb2de9f30a1041d4c6d24a276ee35ae474a6':
  Switch to relative submodules

armintoepfer committed:
Merge pull request #25 in SAT/unanimity from unify-build-script to master

* commit '9fa22b4fbdeb34a97d0e4dbd1ea8611424545f7e':
  Simplify retrieval of external git repos

armintoepfer committed:
htslib submodule switch

Switch back to branch pb for github interoperability

armintoepfer committed:
Merge pull request #26 in SAT/unanimity from htslib-submodule to master

* commit 'db9a86e8584d71d979fb43c9212a1be470f68a73':
  htslib submodule switch

pb-cdunn commented on bamboo.sh:
```diff
@@ -44,7 +44,7 @@ GetGHRepo pbcore _deps
 
 echo "## Fetch submodules"
-git submodule update --init --remote --depth 1
+git submodule update --init --remote
```
What happened? `--depth 1` now breaks **pb-git** too? Did you rebase a submodule?
